examples/compile-time-oil/OilVectorOfStrings.cpp

```diff
@@ -32,7 +32,7 @@ int main() {
   foo.strings.push_back("consectetur adipiscing elit,");
 
   size_t size = -1;
-  int ret = ObjectIntrospection::getObjectSize<Foo>(&foo, &size);
+  int ret = ObjectIntrospection::getObjectSize(foo, size);
```

Comment:
I'm not sure what the right thing to do is here, just pointing it out for discussion...

Making the size argument a reference instead of a pointer means it's less clear to users of the API that it will be modified

Reply:
Agreed. I wasn't sure about this change. On the one hand, a non-const reference is more accurate as the API requires an always valid pointer to something which it then writes to, and adding a distinct error condition for giving it a null pointer seems pointless. On the other hand, non-const references for read only stuff are kind of ingrained. However, if it should be changed it should definitely be done now with the `T &` change.

Reply:
🦀 - `get_object_size(&obj, &mut size);` solves this nicely, though you wouldn't be using a return arg there.

Reply:
In C++ it is more idiomatic to accept a mutable reference. If you use a pointer, you need to deal with `nullptr`.

But it is even more idiomatic to *return* the value. You can use `folly::Expected` or something similar, which is more self-documenting. Or an exception.

Reply:
Yes, I think if `std::expected` was in I'd be using that. Unfortunately exceptions are sort of out as both the `OIL_UNINITIALISED` result and the `OIL_SUCCESS` result need to be on the fast path in different situations, and exceptions would make `OIL_SUCCESS` faster. Thanks for the input!